auth/ante.go

```diff
@@ -14,7 +14,7 @@ import (
 
 // NewAnteHandler returns an AnteHandler that checks signatures,
 // confirm signatures, and increments the feeAmount
-func NewAnteHandler(utxoMapper utxo.Mapper, feeUpdater utxo.FeeUpdater) sdk.AnteHandler {
+func NewAnteHandler(utxoMapper utxo.Mapper, rootchainAddress types.GetAddress, feeUpdater utxo.FeeUpdater) sdk.AnteHandler {
```

Comment:
Why is this a function? Why can't u just pass in a common.Address?

Reply:
At the time SetAnteHandler is called, rootchainAddress has not been set. rootchainAddress gets set in InitChainer which is after the baseapp has been sealed